pkg/habitat_client/client.go

```diff
@@ -70,31 +77,73 @@ func (c *Client) ReadResponse() (*ctl.ResponseWrapper, error) {
 	return &res, nil
 }
 
-// SendRequest sends a request to the default address of the habitat service
-func SendRequest(req interface{}) (*ctl.ResponseWrapper, error) {
-	return SendRequestToAddress(HabitatServiceAddr, req)
+func PostRequest(req, res interface{}, route string) (error, error) {
+	return PostRequestToAddress(fmt.Sprintf("http://%s/%s", HabitatServiceAddr, route), req, res)
 }
 
-func SendRequestToAddress(addr string, req interface{}) (*ctl.ResponseWrapper, error) {
-	client, err := NewClient(addr)
+// PostRequestToAddress posts to the Habitat API. The first error returned is
+// if for if the client is somehow unable to successfully make the request.
+// The second error is if the request is successfully made, but the server responds
+// with an error.
+func PostRequestToAddress(address string, req, res interface{}) (error, error) {
+	reqBody, err := json.Marshal(req)
 	if err != nil {
-		fmt.Println("Error: couldn't connect to habitat service")
-		return nil, err
+		return fmt.Errorf("error marshaling POST request body: %s", err), nil
 	}
 
-	reqWrapper, err := ctl.NewRequestWrapper(req)
+	r, err := http.Post(address, "application/json", bytes.NewReader(reqBody))
 	if err != nil {
-		return nil, err
+		return err, nil
+	}
+
+	resBody, err := io.ReadAll(r.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response body: %s", err), nil
+	}
+	// The request was fine, but we got an error back from server
+	if r.StatusCode != http.StatusOK {
+		return nil, errors.New(string(resBody))
+	}
+
+	err = json.Unmarshal(resBody, res)
+	if err != nil {
+		return fmt.Errorf("error unmarshaling response body into result struct: %s", err), nil
+	}
+
+	return nil, nil
+}
+
+func PostLibP2PRequestToAddress(addr ma.Multiaddr, route string, peerID peer.ID, req, res interface{}) (error, error) {
```

Comment:
nit: might be cleaner if we had a `LibP2PRequest` struct for readability

Reply:
Some of my compass stuff makes this take in less stuff